### ISSUE TITLE
`type_repetition_in_bounds` misses `for<'a>` when combining higher-rank trait bounds

### ISSUE THREAD

**Issue by benwh1** [rust-lang/rust-clippy#9336]
### Summary

Clippy omits the `for<'a>` part of a higher-rank trait bound when trying to combine two trait bounds.

Also happens on 1.63.0 stable.

### Reproducer

I tried this code:

```rust
pub trait Foo<T> {}

pub struct Bar<S, T>(S, T)
where
    S: Foo<T>,
    for<'a> S: Foo<&'a T>;

fn main() {}
```

I expected to see this happen:

```rust
warning: this type has already been used as a bound predicate
 --> src/main.rs:6:5
  |
6 |     for<'a> S: Foo<&'a T>;
  |     ^^^^^^^^^^^^^^^^^^^^^
  |
  = note: requested on the command line with `-W clippy::type-repetition-in-bounds`
  = help: consider combining the bounds: `S: Foo<T> + for<'a> Foo<&'a T>`
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#type_repetition_in_bounds
```

or maybe,

```rust
  = help: consider combining the bounds: `for<'a> S: Foo<T> + Foo<&'a T>`
```

Instead, this happened:

```rust
  = help: consider combining the bounds: `S: Foo<T> + Foo<&'a T>`
```

### Version

```text
rustc 1.65.0-nightly (29e4a9ee0 2022-08-10)
binary: rustc
commit-hash: 29e4a9ee0253cd39e552a77f51f11f9a5f1c41e6
commit-date: 2022-08-10
host: x86_64-unknown-linux-gnu
release: 1.65.0-nightly
LLVM version: 14.0.6
```

### Additional Labels

@rustbot label +I-suggestion-causes-error
